plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/OAuthDetailView.java

```diff
@@ -0,0 +1,364 @@
+				details.getRequestTokenLink();
+			}
+		}
+
+		@Override
+		public void widgetSelected(SelectionEvent e) {
+			BusyIndicator.showWhile(Display.getDefault(), new Runnable(){      
+				public void run(){ 
+					try {
+						AuthDetailsJob job = null;
+						if(StringUtils.isBlank(link)) {
+							job = new AuthDetailsJob();
+							job.setPriority(Job.SHORT);
+							job.schedule();
+						}
+						OAuthDialog dialog = new OAuthDialog(shell, link);
+						job.addJobChangeListener(dialog);
+						dialog.open();
+					}catch(Exception e) {
+						showErrorDialog(shell,e);
+					}
+				}
+			});
+		}
+
+	}
+	
+	private void showErrorDialog(final Shell shell, final Throwable e) {
+		Display.getDefault().asyncExec(new Runnable() {
+			
+			@Override
+			public void run() {
+				ErrorDialog.openError(shell, 
+						"Request Authentication Details Error", 
+						"Unable to retrieve authentication details", ValidationStatus.error(e.getMessage(),e));
+				
+			}
+		});
+	}
+	
+	private class AuthDetailsJob extends Job{
+		
+		private IAuthorizationDetails details;
+		
+		public AuthDetailsJob() {
+			super("RetrieveAuthorizationDetailsJob");
+		}
+
+		public IAuthorizationDetails getDetails() {
+			return details;
+		}
+
+		@Override
+		protected IStatus run(IProgressMonitor monitor) {
+			try {
+				IClient client = new ClientFactory().create(pageModel.getHost(), OpenShiftCoreUIIntegration.getInstance().getSSLCertificateCallback());
+				details = client.getAuthorizationDetails(pageModel.getHost());
+			}catch(Exception e) {
+				return new Status(IStatus.ERROR, OpenShiftUIActivator.PLUGIN_ID, "Unable to retrieve the authentication details", e);
+			}
+			return ValidationStatus.OK_STATUS;
+		}
+		
+	}
+	
+	private class BearTokenAuthenticationProvider implements IConnectionAuthenticationProvider{
+
+		@Override
+		public IConnection update(IConnection conn) {
+			Assert.isLegal(conn instanceof Connection);
+			final Connection connection = (Connection) conn;
+			// might be called from job, switch to display thread to access observables
+			Display.getDefault().syncExec(new Runnable() {
+				
+				@Override
+				public void run() {
+					connection.setAuthType(IAuthorizationContext.AUTHSCHEME_OAUTH);
+					connection.setToken((String)tokenObservable.getValue());
+					connection.setRememberPassword((boolean)rememberTokenObservable.getValue());
+				}
+			});
+			return connection;
+		}
+		
+	}
+	
+	private class OAuthDialog extends Dialog implements IJobChangeListener{
+
+		private String url;
+		private Browser browser;
+		
+		OAuthDialog(Shell parentShell, String url) {
+			super(parentShell);
+			this.url = url;
+		}
+
+
+//		@Override
+//		protected Button createButton(Composite parent, int id, String label,
+//				boolean defaultButton) {
+//			return null;
+//		}
+
+		@Override
+		protected boolean isResizable() {
+			return true;
+		}
+
+
+		@Override
+		protected Point getInitialSize() {
+			return new Point(500, 700);
```

Comment:
> [11:00:49 PM] Max Rydahl Andersen: 1280/1400 is the norm, but most presentation systems is max 1024x

So i guess that if there's a way to use less vertically size it would be preferrable but if not possible, nevermind.